MapboxCoreNavigation/RouteController.swift

```diff
@@ -154,13 +154,26 @@ extension RouteController: CLLocationManagerDelegate {
         lastUserDistanceToStartOfRoute = Double.infinity
     }
     
-    public func userIsOnRoute(_ location: CLLocation) -> Bool {
-        // Find future location of user
+    
+    /**
+     Returns the calculated reroute distance.
+     */
+    public func getRerouteRadius(_ horizontalAccuracy: CLLocationAccuracy) -> CLLocationDistance {
+        return min(RouteControllerMaximumDistanceBeforeRecalculating, horizontalAccuracy + RouteControllerUserLocationSnappingDistance)
+    }
+    
+    /**
+     Takes a `CLLocation` and returns a `CLLocation` infront of the provided location given the current speed.
+     */
+    public func getDeadReckoningLocation(_ location: CLLocation) -> CLLocation {
```

Comment:
A method name only begins with `get` if it starts an asynchronous network request or similar. [A normal getter starts with a noun phrase](https://swift.org/documentation/api-design-guidelines/#name-according-to-side-effects).

This method has no dependencies on RouteController, so it would be cleaner to put it in an extension on CLLocation, as a method called `advanced(by:)` that takes a `TimeInterval`.